tests/ui/components/filters/filteroptionscomponent.js

```diff
@@ -657,4 +658,42 @@ describe('filter options component', () => {
       expect(() => COMPONENT_MANAGER.create('FilterOptions', config)).toThrow();
     });
   });
+
+  describe('interaction with global storage', () => {
+    let globalStorage;
+    let COMPONENT_MANAGER;
+
+    beforeEach(() => {
+      const mockCore = {
+        setStaticFilterNodes: setStaticFilterNodes,
+        setLocationRadiusFilterNode: () => { },
+        filterRegistry: {
+          setStaticFilterNodes: setStaticFilterNodes
+        },
+        globalStorage: new GlobalStorage(),
+        persistentStorage: new PersistentStorage()
+      };
+
+      COMPONENT_MANAGER = mockManager(mockCore);
+
+      defaultConfig = {
+        container: '#test-component',
+        control: 'singleoption',
+        options: options,
+        label: filterOptionsLabel
+      };
+
+      globalStorage = COMPONENT_MANAGER.core.globalStorage;
+    });
+
+    it('deletes the global storage FilterOptions key when constructed', () => {
```

Comment:
This test should help Oliver as he continues on in his refactor. But, I'm not sure how useful the test is once the refactor is complete. We're testing essentially one line of source code here. That line is a fairly low-level implementation detail. Ideally, we would test some higher-level behavior of the component and in so doing, verify a particular storage interaction. For example, Oliver wrote a test to make sure the `QuestionSubmissionComponent` is hidden while a search is loading. The test has the side effect of verifying that the component correctly receives `UNIVERSAL_RESULTS` storage updates. Is there some higher-level behavior we could test here that would have the side effect of verifying the `delete`?

Reply:
I can test that the clear button of the search component deletes the search offset from global storage, so I'll close this out